packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,608 @@
+                       std::string &video_format)
+    : VideoPlayer(messenger),
+      native_window_(native_window),
+      video_format_(video_format) {}
+
+PlusPlayer::~PlusPlayer() { Dispose(); }
+
+int64_t PlusPlayer::Create(const std::string &uri, int drm_type,
+                           const std::string &license_server_url,
+                           bool is_prebuffer_mode) {
+  LOG_INFO("[PlusPlayer] Create player.");
+  if (!video_format_.compare("dash")) {
+    player_ = plusplayer::PlusPlayer::Create(plusplayer::PlayerType::kDASH);
+  } else {
+    player_ = plusplayer::PlusPlayer::Create();
+  }
+
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Fail to create player.");
+    return -1;
+  }
+
+  if (!player_->Open(uri)) {
+    LOG_ERROR("[PlusPlayer] Fail to open uri :  %s.", uri.c_str());
+    return -1;
+  }
+
+  char *appId = nullptr;
+  int ret = app_manager_get_app_id(getpid(), &appId);
+  if (ret != APP_MANAGER_ERROR_NONE) {
+    LOG_ERROR("[PlusPlayer] app_manager_get_app_id failed : %s.",
+              get_error_message(ret));
+    return -1;
+  }
+  player_->SetAppId(std::string(appId));
+  free(appId);
+
+  player_->RegisterListener(this);
+
+  if (drm_type != 0) {
+    if (!SetDrm(uri, drm_type, license_server_url)) {
+      LOG_ERROR("[PlusPlayer] Failed to set drm.");
+      return -1;
+    }
+  }
+
+  if (!SetDisplay()) {
+    LOG_ERROR("[PlusPlayer] Failed to set display.");
+    return -1;
+  }
+
+  SetDisplayRoi(0, 0, 1, 1);
+
+  if (is_prebuffer_mode) {
+    player_->SetPrebufferMode(true);
+    is_prebuffer_mode_ = true;
+  }
+
+  if (!player_->PrepareAsync()) {
+    LOG_ERROR("[PlusPlayer] Failed to prepare");
+    return -1;
+  }
+  return SetUpEventChannel();
+}
+
+void PlusPlayer::Dispose() {
+  LOG_INFO("[PlusPlayer] Disposing.");
+
+  if (player_) {
+    player_->Stop();
+    player_->Close();
+  }
+
+  // drm should be released after destroy of player
+  if (drm_manager_) {
+    drm_manager_->ReleaseDrmSession();
+  }
+}
+
+void PlusPlayer::SetDisplayRoi(int32_t x, int32_t y, int32_t width,
+                               int32_t height) {
+  plusplayer::Geometry roi;
+  roi.x = x;
+  roi.y = y;
+  roi.w = width;
+  roi.h = height;
+  if (!player_->SetDisplayRoi(roi)) {
+    LOG_ERROR("[PlusPlayer] SetDisplayRoi failed.");
+  }
+}
+
+void PlusPlayer::Play() {
+  LOG_INFO("[PlusPlayer] Play video.");
+  plusplayer::State state = player_->GetState();
+  if (state < plusplayer::State::kTrackSourceReady) {
+    LOG_ERROR("[PlusPlayer] Player is not ready.");
+    return;
+  }
+
+  if (state <= plusplayer::State::kReady) {
+    if (!player_->Start()) {
+      LOG_ERROR("[PlusPlayer] Fail to start.");
+    }
+  } else if (state == plusplayer::State::kPaused) {
+    if (!player_->Resume()) {
+      LOG_ERROR("[PlusPlayer] Fail to resume playing.");
+    }
+  }
+}
+
+bool PlusPlayer::SetActivate() {
+  if (!player_->Activate(plusplayer::kTrackTypeVideo)) {
```

Comment:
According to the doccument, it seems that several interfaces need to check kNone state. If the kNone check is unnecessary (logically), there is no need to do it. However, it needs to check whether this can cause errors later.
```
The player state can be all of #State except #State::kNone
```

Reply:
`kNone` state is unnecessary to check according to our code logic, because after create instance of plusplayer(` player_ = plusplayer::PlusPlayer::Create();`), the state change to kIdle from kNone. If create failed(state is Knone), player will not going on.